veps/sig-storage/dic-on-heterogeneous-cluster/dic-on-heterogeneous-cluster.md

```diff
@@ -0,0 +1,656 @@
+the HyperConverged CR, there will be already an existing `DataSource` CR for the specific image. SSP will modify the
+existing `DataSource` CR by replacing the existing `spec.source.pvc` or `spec.source.snapshot` fields, with the new 
+`spec.source.dataSource` field.
+
+CDI, when reconciling a `DataSource` CR, with the `spec.source.dataSource` field populated, will set the `status.source`
+to be the same as the `status.source` of the pointed `DataSource` CR.
+
+For example if the default `DataSource` CR is `centos-stream9-amd64`, the architecture agnostic `DataSource` CR will
+look like this:
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataSource
+metadata:
+  name: centos-stream9 # <<=== the architecture agnostic name
+  namespace: kubevirt-os-images
+spec:
+  source:
+    dataSource:
+      name: centos-stream9-amd64
+      namespace: kubevirt-os-images
+status:
+  conditions:
+  - message: DataSource is ready to be consumed
+    reason: Ready
+    status: "True"
+    type: Ready
+  source:
+    snapshot: # <<=== the actual source, as defined in the *default* DataSource, pointed by this DataSource
+      name: centos-stream9-e065d9079064
+      namespace: kubevirt-os-images
+```
+
+### Importing the Image
+When the cdi-importer creates a `DataVolume` CR importing an image to populate a `VolumeSnapshot` (or the `PVC`), it
+will use the architecture specified in the `spec.template.spec.source.registry.platform.architecture` field of the
+`DataImportCron` CR, if provided.
+
+The importer will pull the image corresponding to the specified architecture and loads it into the `VolumeSnapshot`.
+1. if the `pullMethod` is `pod`, then the cdi-importer will use the platform field to pull the architecture specific
+   image.
+   > **Note**: when the cdi-importer is trying to pull an architecture specific image in a registry, it will set the
+   > `os` field of the image platform to `linux`. The platform reflecting the runtime environment rather than the image
+   > contents (i.e., VM images in KubeVirt may represent different operating systems).
+
+2. when the `pullMethod` is `node`, the cdi-controller creates a pod with an init container with the required image
+   letting the node's CRI to perform the actual pull. In this case the CRI will pull the image matches to its own 
+   architecture. the cdi-controller will make sure that the pod will run on a node with the required architecture by
+   modifying the pod's `nodeSelector` or `nodeAffinity` fields, to force scheduling on a node with the
+   `kubernetes.io/arch` label, matching the architecture specified in the
+   `spec.template.spec.source.registry.platform.architecture` field.
+
+   if there is no available node with the required architecture, the cdi-controller will report this failure in a
+   condition on the corresponding `DataImportCron` CR.
+
+If the image is not a manifest or does not include the specified architecture, the CDI will report this failure in a
```

Comment:
What if it is a single image (no manifest) but the correct arch?

Reply:
Then its DICT must not have the `ssp.kubevirt.io/dict.architectures` annotation.